camayoc/qcs_models.py

```diff
@@ -5,30 +5,53 @@
 import uuid
 
 from pprint import pformat
+from urllib.parse import urljoin
 
+from camayoc import api
 from camayoc.constants import MASKED_PASSWORD_OUTPUT
+from camayoc.constants import (
+    QCS_CREDENTIALS_PATH,
+    QCS_PROFILES_PATH,
+)
 
 
 class QCSObject(object):
     """A base class for other QCS models."""
 
     def __init__(
             self,
+            client=None,
             name=None,
             _id=None):
         """Provide shared methods for QCS model objects."""
         self.name = str(uuid.uuid4()) if not name else name
         self._id = _id
+        self.client = client if client else api.Client()
+        self.endpoint = ''
 
     def fields(self):
-        """Return a dictionary with all fields."""
+        """Return a dictionary with all fields.
+
+        The fields are all data items that may be returned when a `GET` request
+        is sent to the endpoint. It excludes items specific to camayoc such as
+        the client and endpoint associated with objects of this type.
+        """
         fields = self.payload()
         fields['id'] = self._id
         return fields
 
+    def path(self):
+        """Return the path to this object on the server.
+
+        This concatenates the endpoint and the id of this object,
+        to be used to read details about this object or to delete it.
+        """
+        return urljoin(self.endpoint, '{}/'.format(self._id))
+
     def payload(self):
         """Return a dictionary for POST or PUT requests."""
-        return {k: v for k, v in self.__dict__.items() if k != '_id'}
+        def good_key(k): return k not in ['_id', 'client', 'endpoint']
```

Comment:
It seems that a function here is a missile for a bee (if that makes sense).

You could rewrite like the following:

```python
return {
    k: v for k, v in vars(self)
    if k not in ['_id', 'client', 'endpoint']
}

Reply:
Ok that makes sense, I just was not sure how to break up the dict comprehension over several lines